tests/integration/utils/mock-api-vitest.ts

```diff
@@ -73,6 +73,13 @@ export const startMockApi = ({ routes, silent }: MockApiOptions): Promise<MockAp
       }
       addRequest(requests, req)
       res.status(status)
+
+      if (typeof response === 'function') {
```

Comment:
I've added this as a way to define mock routes with dynamic behaviour, rather than having to provide a static value upfront. I'm using this to proxy any requests to the `/blobs` endpoints to the local Blobs server, but I can see it being useful in other situations too.